src/libraries/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/CompositionContainerTests.cs

```diff
@@ -2987,5 +2988,107 @@ public class PartWithRequiredImport
             [Import]
             public object Import { get; set; }
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        public async Task ThreadSafe_CompositionContainer_Calling_Compose_And_GetExport()
+        {
+            var catalog = new AssemblyCatalog(typeof(Export1).Assembly);
+
+            for (int i = 0; i < 100; i++)
+            {
+                var container = new CompositionContainer(catalog, CompositionOptions.IsThreadSafe);
```

Comment:
Without `IsThreadSafe` does this test fail (at least periodically)?

Reply:
Yes it fails with CompositionOptions.Default